packages/communication-react/src/mergedHooks.ts

```diff
@@ -143,3 +143,45 @@ export const usePropsFor = <Component extends (props: any) => JSX.Element>(
     throw 'Could not find props for this component, ensure the component is wrapped by appropriate providers.';
   }
 };
+
+export const useChatPropsFor = <Component extends (props: any) => JSX.Element>(
+  component: Component
+): ComponentProps<Component> => {
+  const chatSelector = getChatSelector(component);
+  const chatProps = useChatSelector(chatSelector);
+  const chatHandlers = useChatHandlers<Parameters<Component>[0]>(component);
+
+  if (chatProps) {
+    if (!chatHandlers) {
+      throw 'Please initialize chatClient and chatThreadClient first!';
+    }
+    return { ...chatProps, ...chatHandlers } as any;
+  }
+
+  if (!chatSelector) {
+    throw "Can't find corresponding selector for this component. Please check the supported components from Azure Communication UI Feature Component List.";
+  } else {
+    throw 'Could not find props for this component, ensure the component is wrapped by appropriate providers.';
+  }
+};
+
+export const useCallingPropsFor = <Component extends (props: any) => JSX.Element>(
```

Comment:
We shouldn't need to recreate these here, we can reexport them directly from the packlets
https://github.com/Azure/communication-ui-library/blob/2cd5e81759120691e969ee028ecc3e7e9c1d6692/packages/calling-component-bindings/src/index.ts#L50